sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexClient.java

```diff
@@ -0,0 +1,354 @@
+     * @return response containing a document object
+     */
+    public Response<Document> getDocumentWithResponse(String key,
+                                                      List<String> selectedFields,
+                                                      RequestOptions requestOptions,
+                                                      Context context) {
+        Mono<Response<Document>> results = asyncClient.getDocumentWithResponse(key,
+            selectedFields,
+            requestOptions,
+            context);
+        return results.block();
+    }
+
+    /**
+     * Suggests documents in the Azure Cognitive Search index that match the given partial query text.
+     *
+     * @param searchText search text
+     * @param suggesterName suggester name
+     * @return A {@link PagedIterableBase} that iterates over {@link SuggestResult} objects
+     * and provides access to the {@link SuggestPagedResponse} object for each page containing
+     * HTTP response and coverage information.
+     */
+    public PagedIterableBase<SuggestResult, SuggestPagedResponse> suggest(String searchText, String suggesterName) {
+        return new PagedIterableBase<>(asyncClient.suggest(searchText,
+            suggesterName, null, null));
+    }
+
+    /**
+     * Suggests documents in the Azure Cognitive Search index that match the given partial query text.
+     *
+     * @param searchText search text
+     * @param suggesterName suggester name
+     * @param suggestOptions suggest options
+     * @param requestOptions additional parameters for the operation.
+     * Contains the tracking ID sent with the request to help with debugging
+     * @param context additional context that is passed through the Http pipeline during the service call
+     * @return A {@link PagedIterableBase} that iterates over {@link SuggestResult} objects
+     * and provides access to the {@link SuggestPagedResponse} object for each page containing
+     * HTTP response and coverage information.
+     */
+    public PagedIterableBase<SuggestResult, SuggestPagedResponse> suggest(String searchText,
+                                                                          String suggesterName,
+                                                                          SuggestOptions suggestOptions,
+                                                                          RequestOptions requestOptions,
+                                                                          Context context) {
+        return new PagedIterableBase<>(asyncClient.suggest(searchText,
+            suggesterName, suggestOptions, requestOptions, context));
+    }
+
+    /**
+     * Sends a batch of document write to the Azure Cognitive Search index.
+     *
+     * @param batch batch of documents to send to the index with the requested action
+     * @return document index result
+     */
+    public IndexDocumentsResult index(IndexBatch<?> batch) {
```

Comment:
@navalev There needs to be documentation here (and in upload/merge/delete/etc.) that mentions `IndexBatchException` and describes how to use it properly. See the .NET SDK for reference: 

https://github.com/Azure/azure-sdk-for-net/blob/8f8b4f5452c0521bfb71636730c947000e44b5f8/sdk/search/Microsoft.Azure.Search.Data/src/Customizations/Documents/IDocumentsOperations.Customization.cs#L393